scm/src/test/java/com/sandboni/core/scm/resolvers/cli/GitDiffRunnerTest.java

```diff
@@ -113,24 +107,33 @@ public void testDiff() throws IOException, SourceControlException {
 
     }
 
+    @Test
+    public void testDiffWithSameToAndFrom() throws Exception {
+        RevisionScope<ObjectId> scope = revisionResolver.resolve("60075ec4", "60075ec4"); // same
+        PowerMockito.doNothing().when(GitDiffRunner.class, "buildDiffCommandHelper", Mockito.any());
+        List<Diff> diffs = GitDiffRunner.diff(repository, scope);
+        assertNotNull(diffs);
+        assertEquals(33, diffs.size());
+    }
+
     private List<Diff> findDiff(List<Diff> diffs, String fileName) {
         return diffs.stream().filter(d -> d.getFromFileName().contains(fileName) || d.getToFileName().contains(fileName)).collect(Collectors.toList());
     }
 
     @Test(expected = SourceControlException.class)
-    public void testDiff_invalidDiffResult() throws IOException, SourceControlException {
-        mockStatic(ProcessRunner.class);
-
-        List<String> diffLines = ResourceFileUtils.getResourceFileContentAsList(getClass(), "gitDiffInvalid.out");
-
-        when(ProcessRunner.runCommand(any(), any())).thenReturn(diffLines);
-
+    public void testDiff_invalidDiffResult() throws SourceControlException {
         RevisionScope<ObjectId> scope = revisionResolver.resolve("44444444", "55555555");
         GitDiffRunner.diff(repository, scope);
-
         assertTrue(true);
     }
 
-
-
+    @Test
+    public void testGitDiffCommandForFeatureFilter() throws SourceControlException {
```

Comment:
it's actually for both java and feature, but I would call this method something like testGitDiffCommand_FileTypeFilterIsAdded()